pkg/api/handlers/libpod/images.go

```diff
@@ -668,3 +670,36 @@ func ImagesRemove(w http.ResponseWriter, r *http.Request) {
 		utils.Error(w, http.StatusInternalServerError, errorhandling.JoinErrors(rmErrors))
 	}
 }
+
+func ImageScp(w http.ResponseWriter, r *http.Request) {
+	decoder := r.Context().Value(api.DecoderKey).(*schema.Decoder)
+	query := struct {
+		Source      string `schema:"source"`
+		Destination string `schema:"destination"`
+		Quiet       bool   `schema:"quiet"`
+	}{
+		// This is where you can override the golang default value for one of fields
+	}
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		utils.Error(w, http.StatusBadRequest, errors.Wrapf(err, "failed to parse parameters for %s", r.URL.String()))
+		return
+	}
+
+	args := []string{query.Source}
+	if len(query.Destination) > 0 {
+		args = append(args, query.Destination)
+	}
+
+	rep, source, dest, _, err := domainUtils.ExecuteTransfer(args, []string{}, query.Quiet)
```

Comment:
@cdoern It appears in the ExecuteTransfer() call tree, that reading from stdin is required. There is no support there for reading from the remote host. The user is queried for the password when no key is configured.

Reply:
@jwhonce the only time where stdin is needed is the intra machine user transfer. This use case is not supported on the remote client for a bunch of reasons. Given that, I think i can keep this structure as is.